service/controller/cloudconfig/cloud_config.go

```diff
@@ -114,3 +130,84 @@ func newCloudConfig(template string, params k8scloudconfig.Params) (string, erro
 
 	return cloudConfig.String(), nil
 }
+
+func SearchCluster(ctx context.Context, ctrlClient ctrl.Client, clusterID, namespace string) (Cluster, error) {
+	var cluster Cluster
+
+	keys := []struct {
+		RandomKey *RandomKey
+		Type      Key
+	}{
+		{RandomKey: &cluster.APIServerEncryptionKey, Type: EncryptionKey},
+	}
+
+	for _, k := range keys {
+		err := search(ctx, ctrlClient, k.RandomKey, clusterID, namespace, k.Type)
+		if err != nil {
+			return Cluster{}, microerror.Mask(err)
+		}
+	}
```

Comment:
Is there going to be more elements in `keys` in the future or could we drop this indirection and inline `search()` here?

Reply:
I have no idea, tbh. I tried to keep it more or less like it was before, but I don't think the current implementation makes much sense now.

Reply:
Your call, either way. :slightly_smiling_face: